### ISSUE TITLE
CORS not working correctly on oauth2 token requests

### ISSUE THREAD

**Issue by danisetg** [yiisoft/yii2#17130]
Hi, I am building an Api restful using yii2, and I have installed the vendor for managing the oauth2 authentication, but I am having some troubles with the login. When I try it from insomnia everything goes great, But when I try it out from my angular 7 front app, I receive the error:
Access to XMLHttpRequest at 'http://localhost:8080/oauth2/token' from origin 'http://localhost:4200' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
Cross-Origin Read Blocking (CORB) blocked cross-origin response http://localhost:8080/oauth2/token with MIME type application/json. See https://www.chromestatus.com/feature/5629709824032768 for more details.
`This is my behavior function in SiteController:

```php
public function behaviors() { $behaviors = parent::behaviors(); unset($behaviors['authenticator']); return [ 'corsFilter' => [ 'class' => \yii\filters\Cors::class, 'cors' => [ // restrict access to 'Origin' => ['*'], // Allow only POST and PUT methods 'Access-Control-Request-Method' => ['*'], // Allow only headers 'X-Wsse' 'Access-Control-Request-Headers' => ['*'], 'Access-Control-Allow-Headers' => ['Content-Type'], // Allow credentials (cookies, authorization headers, etc.) to be exposed to the browser //'Access-Control-Allow-Credentials' => true, // Allow OPTIONS caching 'Access-Control-Max-Age' => 3600, // Allow the X-Pagination-Current-Page header to be exposed to the browser. 'Access-Control-Expose-Headers' => ['*'], ], ], 'access' => [ 'class' => AccessControl::className(), 'only' => ['logout'], 'rules' => [ [ 'actions' => ['logout'], 'allow' => true, 'roles' => ['@'], ], ], ], 'verbs' => [ 'class' => VerbFilter::className(), 'actions' => [ 'logout' => ['post'], ], ], ]; }
```

Can someone help me with this please.

**Comment by danisetg** [comment]
```
public function behaviors()
    {
        return [
            'corsFilter' => [
                'class' => \yii\filters\Cors::class,
                'cors' => [
                    // restrict access to
                    'Origin' => ['*'],
                    // Allow only POST and PUT methods
                    'Access-Control-Request-Method' => ['*'],
                    // Allow only headers 'X-Wsse'
                    'Access-Control-Request-Headers' => ['*'],
                    // Allow credentials (cookies, authorization headers, etc.) to be exposed to the browser
                    //'Access-Control-Allow-Credentials' => true,
                    // Allow OPTIONS caching
                    'Access-Control-Max-Age' => 3600,
                    // Allow the X-Pagination-Current-Page header to be exposed to the browser.
                    'Access-Control-Expose-Headers' => ['*'],
                ],
            ],
            'access' => [
                'class' => AccessControl::className(),
                'only' => ['logout'],
                'rules' => [
                    [
                        'actions' => ['logout'],
                        'allow' => true,
                        'roles' => ['@'],
                    ],
                ],
            ],
            'verbs' => [
                'class' => VerbFilter::className(),
                'actions' => [
                    'logout' => ['post'],
                ],
            ],
        ];
    }
```
I'd really appreciate any help please.

**Comment by samdark** [comment]
What does the request made from your Angular application looks like (including headers)?

**Comment by danisetg** [comment]
@samdark thanks for the reply, I really appreciate it, but I already solved it, it wasn't a yii problem, but a problem with the vendor for oauth2 authentications I had installed. The vendor wasn't setting the behaviors for the cors configuration.
Nevertheless I would like to know if there is a way to apply the configuration of the cors globally, so that every vendor I install use it, even if it doesn't have it setted in its controllers.
